src/chain/tezos/contracts/HicNFT.ts

```diff
@@ -0,0 +1,232 @@
+import {JSONPath} from 'jsonpath-plus';
+
+import {KeyStore, Signer} from '../../../types/ExternalInterfaces';
+import {TezosNodeReader} from '../TezosNodeReader';
+import {TezosNodeWriter} from '../TezosNodeWriter';
+import {ConseilOperator, ConseilSortDirection, ConseilServerInfo} from '../../../types/conseil/QueryTypes';
+import {ConseilQueryBuilder} from '../../../reporting/ConseilQueryBuilder';
+import {TezosConseilClient} from '../../../reporting/tezos/TezosConseilClient';
+import {MultiAssetTokenHelper} from './tzip12/MultiAssetTokenHelper';
+import {TezosParameterFormat} from '../../../types/tezos/TezosChainTypes';
+import {TezosContractUtils} from './TezosContractUtils';
+
+export namespace HicNFTHelper {
+    /*
+     * Type for representing Hic's FA2 contracts' storage.
+     * @param administrator
+     * @param allTokens
+     * @param ledgerMapId
+     * @param metadataMap
+     * @param operatorsMapId
+     * @param paused
+     * @param tokenMetadataMapId
+     */
+    interface HicStorage {
+        administrator: string;
+        allTokens: number;
+        ledgerMapId: number;
+        metadataMapId: number;
+        operatorsMapId: number;
+        paused: boolean;
+        tokenMetadataMapId: number;
+    }
+
+    export const objktsAddress = 'KT1RJ6PbjHpwc3M5rw5s2Nbmefwbuwbdxton';
+    export const hDaoAddress = 'KT1AFA2mwNUMNd4SsujE1YYp29vd8BZejyKW';
+
+    /*
+     * Objkts FA2 contract storage type
+     */
+    export type ObjktsStorage = HicStorage;
+
+    /*
+     * Get an instance of a Objkts contract's storage by querying a given address
+     * @param server The Tezos node to communicate with
+     * @param address The deployed Timelock contract address
+     */
+    export async function getObjktsStorage(server: string): Promise<ObjktsStorage> {
+        const storageResult = await TezosNodeReader.getContractStorage(server, objktsAddress);
+        return {
```

Comment:
@itamarreif: this code is duplicated between `getObjktsStorage` and `getHDaoStorage`